source/best-practices/zsh.rst

```diff
@@ -0,0 +1,157 @@
+zsh 及其配置
+============
+
+:本节贡献者: |田冬冬|\（作者）、
+             |姚家园|\（作者）
+:最近更新日期: 2021-01-21
+:预计阅读时间: 20 分钟
+
+Linux 下有很多 Shell，最常见的是 bash，此外还有 zsh、csh、ksh。推荐使用 zsh。
+
+zsh 有如下特点：
+
+- 语法基本完全兼容于 bash，一般用户完全体会不到其区别
+- zsh 提供命令补全特性，比 bash 的补全要更好用
+- 可配置性强
+
+安装 zsh
+---------
+
+在终端键入 ``zsh --version``，若显示 zsh 版本号，则表示 zsh
+已安装。否则需要安装 zsh。
+
+自 macOS Catalina (10.15) 开始，macOS 系统默认的 Shell 已由 bash 变为 zsh。
+
+Linux 用户可以使用如下命令安装 zsh::
+
+    # Fedora 用户
+    $ sudo dnf install zsh
+
+    # CentOS 用户
+    $ sudo yum install zsh
+
+    # Ubuntu/Debian 用户
+    $ sudo apt install zsh
+
+通过如下命令设置默认 Shell 为 zsh::
+
+    $ chsh -s $(which zsh)
+
+``chsh`` 命令修改的是登陆后的 Shell，因而需要退出当前用户并重新登陆，
+用户的默认 Shell 就从 bash 变成了 zsh 了。打开新的终端键入
+``echo $SHELL``\ ，查看当前 Shell，会显示 :file:`/bin/zsh`\ 。
+
+zsh 的配置文件为 :file:`~/.zshrc`\ 。因而切换到 zsh 后，
+所有的 Shell 配置都不用写到 :file:`~/.bashrc`\ ，而要写到 :file:`~/.zshrc` 中。
+
+Oh My Zsh
+----------
+
+zsh 稍作配置会更加方便好用，`Oh My Zsh <https://ohmyz.sh/>`__ 是由 Oh My Zsh 社区
+维护的一套 zsh 配置文件，使用起来非常方便。一般用户直接使用该配置即可。
+
+安装 Oh My Zsh::
+
+    $ sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+
+该命令会下载 Oh My Zsh 到 :file:`~/.oh-my-zsh` 目录下，生成默认的 zsh 配置文件 :file:`~/.zshrc`\ ，
+并备份老的配置到类似 :file:`.zshrc.pre-oh-my-zsh` 的文件中。
+
+之后可以根据个人习惯修改配置文件 :file:`.zshrc`\ ，如\
+`设置主题 <https://github.com/ohmyzsh/ohmyzsh#themes>`__\ 
+（如 bira）、\ `启用插件 <https://github.com/ohmyzsh/ohmyzsh#plugins>`__\
+等。
+
+设置主题
+^^^^^^^^^
+
+修改配置文件 :file:`.zshrc` 便可以使用不同的主题::
+
+    ZSH_THEME="bira"
+
+打开新的终端，查看主题效果。
+
+Oh My Zsh 自带了很多主题，位于 :file:`~/.oh-my-zsh/themes`\ 目录下，可以在线\
+`预览主题效果 <https://github.com/ohmyzsh/ohmyzsh/wiki/Themes>`__\ 。
+用户也可以启动\ `外部主题 <https://github.com/ohmyzsh/ohmyzsh/wiki/External-themes>`__\ 。
+
+启用插件
+^^^^^^^^^
+
+一般直接修改配置文件 :file:`.zshrc` 便可启用插件::
+
+    plugins=(
+      sudo
+      extract
+      autojump
+      zsh-autosuggestions
+      zsh-syntax-highlighting
+    )
+
+Oh My Zsh 自带了很多插件，位于 :file:`~/.oh-my-zsh/plugins`\ 目录下，也可以\
+`在线查询 <https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins>`__\ 。
```

Comment:
Look here, this link contain the plugins preview.

Reply:
No, they are different. 

- https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins-Overview
- https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins

Reply:
Look at the right side bar, the second one contains the first one.

Reply:
The right sidebar is a global TOC.

Reply:
Which one do you like?

Reply:
The long plugin list is boring, as I have to click each plugin to know that it does.

In the "plugins overview" page, it's much easier to know that each plugin does.

Reply:
Use https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins-Overview in e4c5003edf75ec4bc8a455c4e73f1460de2395b4